### PR TITLE
LRDecayScheduler callback implemented

### PR DESCRIPTION
**Pull request by santipuch590**
Nowadays it is common practice to decay the learning rate when training Deep Neural Networks, as it helps the network converge to the solution.

For that reason we have decided to implement a custom callback that decays the learning rate with the decay factor (lrDecayScheduler_rate) specified by the user at specific epochs (lrDecayScheduler_epochs), also specified by the user. It also allows to decay the learning rate for all epochs if lrDecayScheduler_epochs is set to None.

We have decided to create this pull request so that other groups can benefit from this training procedure, as we have found out it can improve the overall performance of the network.

### CODE REVIEW

**Comment by lluisgomez** [comment]
Well done!
Thanks.

**Comment by dvazquezcvc** [comment]
Isn't this callback already implemented in Keras? 

https://keras.io/callbacks/
LearningRateScheduler

???

**Comment by santipuch590** [comment]
Yes, I know Keras has its own Learning Rate Scheduler, but there are 2 reasons why I implemented a custom callback: 
1) I wanted to try to implement a custom callback in Keras, 
2) I tried to integrate the LRScheduler callback into the framework but it turned out to be easier to create our own callback adapted to our specific problem, which was decaying the learning rate at specific epochs, and also it was easier to provide configuration options using the custom implementation.

Anyway, thank you for the suggestion, and for accepting the PR.

**Comment by dvazquezcvc** [comment]
:)
